### PR TITLE
Feature/cv2 2734 alerts with buttons

### PR DESCRIPTION
**Pull request by danielevalverde**
## Description

Refactoring to use only one Alert component instead of SucessAlert and WarningAlert components and pass the type of alert to Alert component as prop.
add small button and dismiss props to Alert
and fix tests

Reference: CV2-2734

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## Checklist

- [X] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/cds/alerts-and-prompts/Alert.js
Comment (danielevalverde) [reply src/app/components/cds/alerts-and-prompts/Alert.js]: Thanks for catching :) 
deleted :pray:

**Comment by brianfleming** [comment]
@danielevalverde thanks! any tips or suggestions on how to review the output of this component visually? so i can see all the variations, etc?

**Comment by danielevalverde** [comment]
> @brianfleming 

I think the easiest way is  change the type of alert here: 
by "warning" , "info", "error"
https://github.com/meedan/check-web/blob/feature/CV2-2734-alerts-with-buttons/src/app/components/team/SmoochBot/SmoochBotNewsletterEditor.js#L226

you can also pass the booelan props "dismiss" and the content of small button as the  "button" prop. 
and go to the "settings/tipline" page on the "Newsletter" tab where you will be able to see the Alert 

Happy to make a call and share my screen :)

**Review by brianfleming — CHANGES_REQUESTED** [review]
@danielevalverde nice progress! I have a couple nits and one suggestion.

- [ ] The drop shadow is missing when the bool is provided. It should include `box-shadow: 0 4px 16px rgba(0, 0, 0, 0.1);`
- [ ] It looks like the title is getting conflicting font sizes, should be `subtitle2` but it looks like its being overwritten by multiple type sizes being applied
- [ ] The color of the X dismiss should follow the theme of the alert, so either `validationSecondary`, `errorSecondary`, `alertSecondary`, or `brandSecondary`
- [ ] Can you change the warning icon to be the `report_problem` icon from mat icons? its a triangle instead of cirlce
- [ ] another nit about the icons ^, they should be the same color as the title element
- [ ] when there is a button present, the bottom padding on the alert is too large. This is because the button has a bottom margin, making the space larger. I suggest removing the bottom margin so the padding around the content is consistent:
Screenshot 2023-03-07 at 9 57 12 AM

- [ ] to go along with that previous comment ^, the gap between the button and the content is a bit tight. I think its `8px` now, could you up it to `12px`?
- [ ] could you move the X button a bit so it lines up with the content and other icon, vertically? see the red line below:
Screenshot 2023-03-07 at 9 59 37 AM

- [ ] there should be a "gutter" for the content on the right so, when it wraps it doesnt flow in to the X dismiss button. If you look at this mock up, the designs on the left versus what is in this branch on the right, you will see how the content runs into the X and below it.
Screenshot 2023-03-07 at 10 11 40 AM

### Suggestion:
- thanks for the tip on testing this. Another way to test the "toast" version would be to replace this component with your new one, since we would like to use your new component in this location as well:
Screenshot 2023-03-07 at 9 54 21 AM

Thanks! let me know if you would like to discuss or if I can provide any additional details on these changes.

**Comment by amoedoamorim** [comment]
> ### Suggestion:
> * thanks for the tip on testing this. Another way to test the "toast" version would be to replace this component with your new one, since we would like to use your new component in this location as well:
> 
> Screenshot 2023-03-07 at 9 54 21 AM

@brianfleming Let's leave this change for a new ticket and assess feasibility first as it may not be a trivial change.

**Comment by danielevalverde** [comment]
@brianfleming Thank you for the detailed review :pray: 

I fixed the items you listed, but please let me know if it is still missing something. 

Also: please noticed i change some prop names:
button -> buttonLabel
It is no longer necessary to pass  "dismiss" prop, it is just necessary to pass the "onClose" function, like we have here: 
https://github.com/meedan/check-web/blob/feature/CV2-2734-alerts-with-buttons/src/app/components/team/SmoochBot/SmoochBotNewsletterEditor.js#L227

**Comment by dariusk** [comment]
This looks great overall! I have one piece of feedback to make the code a little easier to read. The `Alert` component accepts a `type` prop, which references an `alertTypes` object that has all the different variables for styles for each alert in it. This is good, but then you call `alertTypes[type].icon`, `alertTypes[type].primaryColor`, etc.

Near the top of the main `Alert` function, right after you call `useStyles`, you could do something like this:

`const {icon, primaryColor, secondaryColor, borderInfo} = alertTypes[type];`

This uses a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#description) in case you aren't familiar with that JS concept. Basically it lets you pull out the named properties of an object (in this case one of the sub-objects of `alertTypes`) into variables of the same name as each property. So now we have `borderInfo`, `primaryColor`, etc, set to variables on the main scope of `Alert`.

Then in the code you could just refer to variables themselves without `alertTypes[type]`, so for example lines 93-100 could look like

```jsx
  return (
    <Box
      className={[classes.box, 'alert'].join(' ')}
      style={{
        background: primaryColor,
        border: hasBorder ? borderInfo : null,
        boxShadow: hasBorder ? '0 4px 16px rgba(0, 0, 0, 0.1)' : null,
      }}
```

...notice how there is just `primaryColor` and `borderInfo` here instead of needing to address the `alertTypes`.

So you could just delete all occurrences of `alertTypes[type]` for clearer code.

**Review by dariusk — CHANGES_REQUESTED** [review]
(See above comment on destructuring for the "official" review)

**Comment by danielevalverde** [comment]
> (See above comment on destructuring for the "official" review)

@dariusk, Thank yoou :heart: 
Applied your review :pray:

**Comment by brianfleming** [comment]
Everything I mentioned looks like it was taken care of @danielevalverde thanks. I didn't get a chance to add another review before this was merged (did you mean to merge without approvals?), anyway, I would have added:

- the dismiss X should have a tooltip so the user knows what will happen if its clicked

**Comment by danielevalverde** [comment]
I merged the progress, but i can keep work on it, Brian :) 
Will add the Tooltip. Thanks :pray:

**Pull request merged**
